### PR TITLE
Remove `Roles` Usage

### PR DESCRIPTION
**Pull request by ryanrath**
## Description
**NOTE: This PR is based off of #799 and therefor #760 and should not be merged until these have been merged. This also the reason that there are so many commits / files.**

**This PR really starts @ https://github.com/ubccr/xdmod/commit/b46d9f0994e20814a0142ad52381a1a05a682b8d**

This PR removes the last remaining usages of the `Roles` table. This involved the following:
- Removing unused functions in `XDUser.php`
- Removing tests for functions removed in `XDUser.php`
- Updated the `XDReportManager::enumerateReportTemplates` function to utilize the `report_template_acls` table which uses `acls` as opposed to `ReportTemplateAcls` which uses `Roles`. 
- Added all the infrastructure for the new `report_template_acls` table.

## Motivation and Context
Since we're phasing out the Roles system there's no reason to keep this unused code around anymore. 

## Tests performed
OpenXDMoD docker ( fresh_install / upgrade):
- Unit Tests
- Component Tests
- Integration Tests
- UI Tests
- Regression Tests

XSEDE Docker:
- Manual testing to make sure that things are working as expected.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [X] Code Refactor / Cleanup

## Checklist:

- [X] My code follows the code style of this project as found in the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

**Comment by smgallo** [comment]
@ryanrath 	Now that the 2 dependent PRs have been merged can we rebase this and clean it up so it's easier to review?

**Comment by ryanrath** [comment]
Absolutely, I'll fix the conflicts / clean it up.

File: configuration/etl/etl.d/acls-import.json [hunk 1]
```diff
@@ -68,7 +68,23 @@
...
+        },
+        {
+            "name": "report-template-acls",
+            "description": "report-template-acls + data",
+            "namespace": "ETL\\Ingestor",
+            "class": "DatabaseIngestor",
+            "options_class": "IngestorOptions",
+            "definition_file": "acls/report-template-acls.json",
+            "endpoints": {
+                "source": {
```
Comment (smgallo) [inline configuration/etl/etl.d/acls-import.json]: Is this necessary? `moddb` is defined in the defaults for this file.

File: configuration/etl/etl_tables.d/acls/report_template_acls.json [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        "is_unique": true
+      },
+      {
+        "name": "idx_report_template_id",
+        "columns": [
+          "report_template_id"
+        ]
+      },
+      {
+        "name": "idx_acl_id",
```
Comment (smgallo) [inline configuration/etl/etl_tables.d/acls/report_template_acls.json]: Is this needed since there is a foreign key constraint below?

**Comment by ryanrath** [comment]
All re-based / fixed.

File: configuration/etl/etl_tables.d/acls/report_template_acls.json (hunk 2 above)
Comment (ryanrath) [reply configuration/etl/etl_tables.d/acls/report_template_acls.json]: Nope, just caught myself up on how MySQL handles FK's `MySQL requires that foreign key columns be indexed; if you create a table with a foreign key constraint but no index on a given column, an index is created`. I'll remove the index.

Comment (ryanrath) [reply configuration/etl/etl_tables.d/acls/report_template_acls.json]: Or, it is I guess. `classes/ETL/DBModel/Table.php@200-203` If an index is not found for a column in a FK constraint then it exceptions out.

Comment (smgallo) [reply configuration/etl/etl_tables.d/acls/report_template_acls.json]: @ryanrath 	Add that as a bug in Asana for 8.5.

Comment (ryanrath) [reply configuration/etl/etl_tables.d/acls/report_template_acls.json]: kk will do

Comment (ryanrath) [reply configuration/etl/etl_tables.d/acls/report_template_acls.json]: added

**Pull request merged**